pkg/ldd/ldso_freebsd.go

```diff
@@ -0,0 +1,48 @@
+package ldd
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+const ldso = "/libexec/ld-elf*.so.*"
+
+func LdSo() (string, error) {
+	n, err := filepath.Glob(ldso)
+	if err != nil {
+		return "", err
+	}
+	if len(n) == 0 {
+		return "", fmt.Errorf("No ld.so matches %v", ldso)
+	}
+	return n[0], nil
+}
+
+// runinterp runs the interpreter with the file as an argument.
+// For each returned line it looks for => as the second field,
+// indicating a real .so (as opposed to the .vdso or a string
+// like 'not a dynamic executable'.
+func runinterp(interp, file string) ([]string, error) {
+	var names []string
+	cmd := exec.Command(interp, file)
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, "LD_TRACE_LOADED_OBJECTS=1")
+	o, err := cmd.Output()
+	if err != nil && o == nil {
+		return nil, err
+	}
+	for _, p := range strings.Split(string(o), "\n") {
```

Comment:
This for loop looks the same in each one. I think we need a common function that postprocesses output, and the per-os part can be the thing that runs ldso.